text/0004-structured_data_io.md

```diff
@@ -83,15 +98,30 @@ reader from scratch & reading it's values:
   if err != nil {
     panic(err)
   }
-  fmt.Println(ent.Value) // "bar"
+  fmt.Println(ent.Value) // {"name":"bar"}
 
   _, err := str.ReadEntry()
   fmt.Println(err.Error()) // EOF
 ```
 
-### Stream
+### Stream & Top Level Data
 A _stream_ refers to refer to both a _reader_ and a _writer_ collectively.
 
+_Top Level_ refers to the first entry in a discrete set of data.
```

Comment:
This is not totally accurate. "Top-level" refers to the specific type of the outermost element in a structured piece of data. It's an orthogonal concept to "entries".

And while I know it's repeated below, I think it would help to specify in this section that Qri requires a top-level type of either array or object.